geps/gep-1619.md

```diff
@@ -218,10 +433,144 @@ The following are items that we intend to resolve before we consider this GEP im
+```go
+// HttpCookie defines a cookie to achieve session persistence.
+//
+// Support: Core
+type HttpCookie struct {
+    // Name defines the cookie's name.
+    //
+    // Support: Core
+    //
+    // +kubebuilder:validation:MaxLength=4096
+    Name String `json:"name,omitempty"`
+    
+    // CookieAttributes defines the cookie's attributes.
+    //
+    // Support: Core
+    // +optional
+    CookieAttributes []CookieAttribute `json:cookieAttributes`
+}
+
+// CookieAttribute defines the cookie's attributes.
+type CookieAttribute map[string][]string
+)
+```
+
+The API could also be a mix of individual fields and listed attributes. More specifically, we could separate the key
+attributes with no value into a list. This approach is taken by [Haproxy Ingress](https://haproxy-ingress.github.io/docs/configuration/keys/#affinity)
+with their `session-cookie-keywords` field. This provides flexibility for simple boolean-typed attributes, while
+validating attributes that have values. However, this approach may be confusing to users as uses two different API
+patterns for cookie attributes.
+
+```go
+// HttpCookie defines a cookie to achieve session persistence.
+//
+// Support: Core
+type HttpCookie struct {
+    // Name defines the cookie's name.
+    //
+    // Support: Core
+    //
+    // +kubebuilder:validation:MaxLength=4096
+    Name String `json:"name,omitempty"`
+
+    // SameSite defines the cookie's SameSite attribute.
+    //
+    // Support: Extended
+    //
+    // +optional
+    // +kubebuilder:validation:Enum=Strict;Lax;None
+    SameSite SameSiteType `json:"sameSite,omitempty"`
+    
+    // Domain defines the cookie's Domain attribute.
+    //
+    // Support: Extended
+    //
+    // +optional
+    // +kubebuilder:validation:MaxLength=4096
+    Domain String `json:"domain,omitempty"`
+    
+    // CookieKeywords defines the cookie's attributes that have no value.
+    //
+    // Support: Extended
+    // +optional
+    CookieKeywords []CookieKeyword `json:cookieKeywords`
+}
+
+// CookieKeyword defines the cookie's attributes that have no value.
+type CookieKeyword string
+
+const (
+    // CookieKeywordsHttpOnly specifies the HttpOnly cookie attribute.
+    CookieKeywordsHttpOnly HttpOnlyMode = "HttpOnly"
+    // CookieKeywordsSecure specifies the Secure cookie attribute.
+    CookieKeywordsSecure HttpOnlyMode = "Secure"
+)
+```
+
+Taking a different approach, this GEP could design a generic approach to configuring load balancing policy. Instead of
+only having a metaresource specifically for session persistence, a metaresource called `LoadBalancerPolicy` of which
+includes a field for session persistence along with other load balancer related configuration.
+
+The `LoadBalancerPolicy` design provides tighter coupling with other load balancing configuration which help reduce CRD
+proliferation, but may limit API flexibility as it forces coupling between concepts that may not be appropriate for
+current and future implementations.
```

Comment:
Agreed, so we should not combine session persistence  and load balancing policy into a single API